src/cdomains/affineEquality/sparseImplementation/sparseVector.ml

```diff
@@ -0,0 +1,393 @@
+open AbstractVector
+open RatOps
+open ConvenienceOps
+
+open BatList
+open Batteries
+
+module List = BatList
+module Array = BatArray
+
+module SparseVector: AbstractVector =
+  functor (A: RatOps) ->
+  struct
+    include ConvenienceOps (A)
+
+    type t = {
+      entries: (int * A.t) list ;
+      len: int
+    }[@@deriving eq, ord, hash]
+
+    let copy v = v 
+
+    let of_list l = 
+      let entries' = List.filteri_map (fun i x -> if x <> A.zero then Some (i, x) else None) l
+      in {entries = entries'; len = List.length l}
```

Comment:
Two comments: Please ident like so

~~~ocaml
    let ... = .... in
~~~

and not with the in in the next line. (I think it's like this almost everywhere).

Also, here field punning (https://dev.realworldocaml.org/records.html) makes to code a little nicer:

```suggestion
      let entries = List.filteri_map (fun i x -> if x <> A.zero then Some (i, x) else None) l in
      {entries; len = List.length l}
```